### PR TITLE
ARROW-3940: [Python/Documentation] Add required packages to the development instruction

### PR DESCRIPTION
**Pull request by sekikn**

### CODE REVIEW

File: python/doc/source/development.rst [hunk 1]
```diff
@@ -125,9 +125,13 @@ dependencies will be automatically built by Arrow's third-party toolchain.
                           libboost-filesystem-dev \
                           libboost-system-dev \
                           libboost-regex-dev \
+                          python-dev \
+                          autoconf \
                           flex \
                           bison
 
+If you are building Arrow for Python 3, install ``python3-dev`` instead of ``python-dev``.
```
Comment (wesm) [inline python/doc/source/development.rst]: If both are installed, does the user need to pass some extra things to cmake to select the one they want (e.g. `-DPYTHON_EXECUTABLE`)?

**Review by wesm — COMMENTED** [review]
Thanks for adding these. One comment about the system Python toolchain

File: python/doc/source/development.rst (hunk 1 above)
Comment (sekikn) [reply python/doc/source/development.rst]: Exactly! I added an explanation about that to the PR. Thanks for the comment.

Comment (sekikn) [reply python/doc/source/development.rst]: Oh, I accidentally closed this. I'll submit it as another PR.

**Pull request closed without merging**